optax/_src/transform.py

```diff
@@ -953,6 +953,57 @@ def update_fn(updates, state, params=None):
   return base.GradientTransformation(init_fn, update_fn)
 
 
+class ScaleByOnlineNewtonStepState(NamedTuple):
+  """State holding the inverse of the sum of gradient outer products to date."""
+  hessian_inv: base.Updates
+
+
+def sherman_morrison(a_inv, u):
+  """Sherman Morrison formula to compute the inverse of the sum of an invertible
```

Comment:
The first line of the docstring should be a <=80 character summary according to https://google.github.io/styleguide/pyguide.html#381-docstrings and followed by a blank line.

In this case you could just keep Sherman Morrison formula as summary and move the rest to the detailed description.

Please also add an Args and Returns section as the internal linter won't allows us to merge non-trivial docstrings (i.e. docstrings that don't just say "Returns X given Y.") without Args and Returns sections.